### PR TITLE
(CDAP-4376) Authorization handler must be enabled only when both auth…

### PR DESCRIPTION
**Pull request by bdmogal**
…entication and authorization are enabled.

Jira: [CDAP-4376](https://issues.cask.co/browse/CDAP-4376)
Build: http://builds.cask.co/browse/CDAP-DUT3916

### CODE REVIEW

File: cdap-cli-tests/src/test/java/co/cask/cdap/cli/AuthorizationCLITest.java [hunk 1]
```diff
@@ -25,9 +26,19 @@
 import co.cask.cdap.proto.security.Action;
 import co.cask.cdap.proto.security.Principal;
 import co.cask.cdap.proto.security.Role;
+import co.cask.cdap.security.server.AbstractAuthenticationHandler;
+import co.cask.cdap.security.server.BasicAuthenticationHandler;
 import co.cask.common.cli.CLI;
+import com.google.inject.Inject;
 import org.apache.twill.filesystem.Location;
 import org.apache.twill.filesystem.LocationFactory;
+import org.eclipse.jetty.security.Authenticator;
```
Comment (anwar6953) [inline cdap-cli-tests/src/test/java/co/cask/cdap/cli/AuthorizationCLITest.java]: why so many new imports in this file

Comment (bdmogal) [inline cdap-cli-tests/src/test/java/co/cask/cdap/cli/AuthorizationCLITest.java]: clearly unused. will fix.

File: cdap-security/src/main/java/co/cask/cdap/security/authorization/AuthorizerInstantiatorService.java [hunk 2]
```diff
@@ -99,6 +102,13 @@ public AuthorizerInstantiatorService(CConfiguration cConf, AuthorizationContextF
   @Override
   protected void startUp() throws Exception {
     if (!authorizationEnabled) {
+      LOG.debug("Authorization is disabled. Using a no-op authorizer.");
+      this.authorizer = new NoOpAuthorizer();
+      return;
+    }
+    if (!authenticationEnabled) {
+      LOG.debug("Authorization is enabled. However, authentication is disabled. So all authorization operations will " +
```
Comment (anwar6953) [inline cdap-security/src/main/java/co/cask/cdap/security/authorization/AuthorizerInstantiatorService.java]: make the log messages consistent.
`Using a no-op authorizer.`
`... all authorization operations will be no-op.`

**Comment by anwar6953** [comment]
LGTM besides 1 comment about log messages.

**Comment by bdmogal** [comment]
Addressed comments - which were only about log messages and unused imports. Ran checkstyle locally. Merging.

**Pull request merged**